python/lsst/rubintv/production/mountTorques.py

```diff
@@ -164,68 +168,77 @@ def plotMountTracking(dataId, butler, client, figure, saveFilename, logger):
     start = time.time()
-
-    # Plotting
-    figure.clear()
-    title = f"Mount Tracking {dataIdString}, Azimuth = {azimuth:.1f}, Elevation = {elevation:.1f}"
-    plt.suptitle(title, fontsize=18)
-    # Azimuth axis
-    plt.subplot(3, 3, 1)
-    ax1 = az['azimuthCalculatedAngle'].plot(legend=True, color='red')
-    ax1.set_title("Azimuth axis", fontsize=16)
-    ax1.axvline(az.index[0], color="red", linestyle="--")
-    ax1.set_xticks([])
-    ax1.set_ylabel("Degrees")
-    plt.subplot(3, 3, 4)
-    plt.plot(fit_times, az_error, color='red')
-    plt.title(f"Azimuth RMS error = {az_rms:.2f} arcseconds")
-    plt.ylim(-10.0, 10.0)
-    plt.xticks([])
-    plt.ylabel("Arcseconds")
-    plt.subplot(3, 3, 7)
-    ax7 = az_torque_1['azimuthMotor1Torque'].plot(legend=True, color='blue')
-    ax7 = az_torque_2['azimuthMotor2Torque'].plot(legend=True, color='green')
-    ax7.axvline(az.index[0], color="red", linestyle="--")
-    ax7.set_ylabel("Torque (motor current in amps)")
-
-    # Elevation axis
-    plt.subplot(3, 3, 2)
-    ax2 = el['elevationCalculatedAngle'].plot(legend=True, color='green')
-    ax2.set_title("Elevation axis", fontsize=16)
-    ax2.axvline(az.index[0], color="red", linestyle="--")
-    ax2.set_xticks([])
-    plt.subplot(3, 3, 5)
-    plt.plot(fit_times, el_error, color='green')
-    plt.title(f"Elevation RMS error = {el_rms:.2f} arcseconds")
-    plt.ylim(-10.0, 10.0)
-    plt.xticks([])
-    plt.subplot(3, 3, 8)
-    ax8 = el_torque['elevationMotorTorque'].plot(legend=True, color='blue')
-    ax8.axvline(az.index[0], color="red", linestyle="--")
-    ax8.set_ylabel("Torque (motor current in amps)")
-
-    # Nasmyth2 rotator axis
-    plt.subplot(3, 3, 3)
-    ax3 = rot['nasmyth2CalculatedAngle'].plot(legend=True, color='blue')
-    ax3.set_title("Nasmyth2 axis", fontsize=16)
-    ax3.axvline(az.index[0], color="red", linestyle="--")
-    ax3.set_xticks([])
-    plt.subplot(3, 3, 6)
-    plt.plot(fit_times, rot_error, color='blue')
-    plt.title(f"Nasmyth RMS error = {rot_rms:.2f} arcseconds")
-    plt.ylim(-100.0, 100.0)
-    plt.subplot(3, 3, 9)
-    ax9 = rot_torque['nasmyth2MotorTorque'].plot(legend=True, color='blue')
-    ax9.axvline(az.index[0], color="red", linestyle="--")
-    ax9.set_ylabel("Torque (motor current in amps)")
-    plt.savefig(saveFilename)
-
-    end = time.time()
-    elapsed = end-start
-    logger.debug(f"Elapsed time for plots = {elapsed}")
-
-    return True
+    if saveFilename is not None:
+        # Plotting
+        figure.clear()
+        title = f"Mount Tracking {dataIdString}, Azimuth = {azimuth:.1f}, Elevation = {elevation:.1f}"
+        plt.suptitle(title, fontsize=18)
+        # Azimuth axis
+        plt.subplot(3, 3, 1)
+        ax1 = az['azimuthCalculatedAngle'].plot(legend=True, color='red')
+        ax1.set_title("Azimuth axis", fontsize=16)
+        ax1.axvline(az.index[0], color="red", linestyle="--")
+        ax1.set_xticks([])
+        ax1.set_ylabel("Degrees")
```

Comment:
Can the x-axes be labeled/given units?  It looks like they're all times, except for the Nasmyth RMS error (which is degrees).

Reply:
They are actually all sharing the bottom axis, and they are indeed all time, which I think is obvious enough at the bottom. However, for some reason I can't quite work out, this plot is showing its axis (which is actually seconds, centered on zero), an so looks different (the bottom is absolute time, the intermediate plots are relative time, centered on the observation midpoint for fitting reasons).

I just don't know why this one is showing its ticks when the others aren't and they're all turned off the same way...

Reply:
Ah, caught it. Removing that random addition of ticks and I think we're done.